domene/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/domene/behandling/Behandling.kt

```diff
@@ -213,6 +218,41 @@ data class Behandling(
             }
         }
 
+    fun oppdaterVurderingsPeriode(behandlingId: BehandlingId, vurderingsperiode: Periode, saksbehandler: Saksbehandler): Either<KanIkkeOppdatereVurderingsperiode, Behandling> {
+        if (!saksbehandler.erSaksbehandler()) {
+            logger.warn { "Navident ${saksbehandler.navIdent} med rollene ${saksbehandler.roller} har ikke tilgang til å endre vurderingsperiode på behandling $behandlingId" }
+            return KanIkkeOppdatereVurderingsperiode.HarIkkeTilgang(
+                kreverEnAvRollene = setOf(Saksbehandlerrolle.SAKSBEHANDLER),
+                harRollene = saksbehandler.roller,
+            ).left()
+        }
+
+        if (!this.vurderingsperiode.inneholderHele(periode = vurderingsperiode)) {
+            logger.warn { "Kan ikke oppdatere vurderingsperiode fordi innsendt periode er utenfor nåværende vurderingsperiode for behandlingId $behandlingId" }
+            KanIkkeOppdatereVurderingsperiode.KanKunKrympe(this.vurderingsperiode).left()
+        }
+
+        val vilkårssett = this.vilkårssett.krymp(vurderingsperiode)
+        val stønadsdager = this.stønadsdager.krymp(vurderingsperiode)
+        return this.copy(
+            vurderingsperiode = vurderingsperiode,
+            vilkårssett = vilkårssett,
+            stønadsdager = stønadsdager,
+        ).right()
+    }
+
+    fun oppdaterUtløsendeVilkår(behandlingId: BehandlingId, saksbehandler: Saksbehandler, utløsendeVilkår: UtløsendeVilkår): Either<KanIkkeUtløsendeVilkår, Behandling> {
```

Comment:
Utløsende vilkår. Dette skal vel kun brukes i brev? Ville det vært tydeligere å kalle det noe nærmere brev? Føles litt fremmed for meg med et vilkårUtenforVurderingsperioden 🤷